shared/js/l10n.js

```diff
@@ -1607,27 +1684,32 @@
       }
     }
 
-    if (!this.ctx.availableLocales.length) {
-      // if there was no availableLanguages meta,
-      // register the default locale only
-      this.ctx.registerLocales(this.ctx.defaultLocale);
+    if (navigator.mozApps && navigator.mozApps.getAdditionalLanguages) {
+      navigator.mozApps.getAdditionalLanguages().then(
+        registerLocales.bind(this, meta));
+    } else {
+      registerLocales.call(this, meta);
     }
-    return initLocale.call(this);
```

Comment:
It separates `extendLocales` from `ctx.registerLocales` from `initLocale`.  It makes `extendLocales` pure which makes it easier to understand and test.  It doesn't conflate registration with initialization, which are two logically separate steps.